agent/php_internal_instrument.c

```diff
@@ -849,6 +879,33 @@ NR_INNER_WRAPPER(mysqli_real_connect) {
   zval* mysqli_obj = NULL;
   int zcaught = 0;
 
+  /* PHP 8.1 and later will report a deprecation warning if null is sent where
+   * a non-null argument value is expected.  For these PHP versions we use the
+   * same argument type specification as the mysqli::real_connect() extension
+   * uses to avoid creating this deprecation warning.
+   * For older PHPs continue to use the same specification string as previously
+   * to minimize any chances of introducing new problems.
+   */
+#if ZEND_MODULE_API_NO >= ZEND_8_1_X_API_NO
+  if (FAILURE
+      == zend_parse_parameters_ex(
+          ZEND_PARSE_PARAMS_QUIET, ZEND_NUM_ARGS() TSRMLS_CC, "o|s!s!s!s!l!s!l",
```

Comment:
I think, in all cases where we are changing the specification string, I would prefer a smaller ifdef- perhaps around a local variable. As it is, I expect some other differences in these blocks besides the sting. Something like this?
```
#if ZEND_MODULE_API_NO >= ZEND_8_1_X_API_NO
char string_spec[] = "o|s!s!s!s!l!s!l";
#else
char string_spec[] = "|s!s!s!s!l!s!l";
#endif
```

Reply:
I could do this for certain cases, but for cases where we have to include `port_is_null` conditionally for 8.1+, (like this one) I'll have to duplicate the zpp calls anyway to optionally include that arg.

Reply:
See the changes in 87ae53c5252fdd79b0eec65646ab893678e16063 that implement this